rmw_microxrcedds_c/src/rmw_init.c

```diff
@@ -198,7 +208,7 @@ rmw_init(const rmw_init_options_t * options, rmw_context_t * context)
   // Micro-XRCE-DDS Client initialization
 
 #ifdef RMW_UXRCE_TRANSPORT_SERIAL
-  int fd = open(context->impl->connection_params.serial_device, O_RDWR | O_NOCTTY);
+  int fd = open(context->impl->transport_params.serial_device, O_RDWR | O_NOCTTY);
```

Comment:
This cannot be working. It should be changed to 
```suggestion
  int fd = open(options->impl->transport_params.serial_device, O_RDWR | O_NOCTTY);
```